modules/frontend/searchsharding.go

```diff
@@ -71,7 +74,7 @@ type SearchSharderConfig struct {
 }
 
 // newSearchSharder creates a sharding middleware for search
-func newSearchSharder(reader tempodb.Reader, o *overrides.Overrides, cfg SearchSharderConfig, sloCfg SLOConfig, logger log.Logger) Middleware {
+func newSearchSharder(reader tempodb.Reader, o *overrides.Overrides, cfg SearchSharderConfig, sloCfg SLOConfig, progress searchProgressFactory, logger log.Logger) Middleware {
```

Comment:
If I'm reading this right it looks like the progress factory is only called once.   I think it would help readability if we passed in the actual `shardedSearchProgress` and let `Roundtrip` set the static fields like total blocks/jobs directly.

Reply:
yup, seems obvious now that you point it out. good call